k4MarlinWrapper/src/components/TrackingCellIDEncodingSvc.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2019-2023 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "k4MarlinWrapper/TrackingCellIDEncodingSvc.h"
+
+// k4fwcore
+#include <k4Interface/IGeoSvc.h>
+
+#include <GaudiKernel/MsgStream.h>
+
+DECLARE_COMPONENT(TrackingCellIDEncodingSvc);
+
+TrackingCellIDEncodingSvc::TrackingCellIDEncodingSvc(const std::string& name, ISvcLocator* svc)
+    : base_class(name, svc), m_geoSvc(svc->service("GeoSvc")) {}
+
+TrackingCellIDEncodingSvc::~TrackingCellIDEncodingSvc() {}
+
+StatusCode TrackingCellIDEncodingSvc::initialize() {
+  try {
+    info() << "Taking the encoding string as specified in dd4hep constant: " << m_encodingStringVariable.value()
+           << endmsg;
+
+    auto encodingString = m_geoSvc->constantAsString(m_encodingStringVariable.value());
```

Comment:
Don't you need to retrieve the DD4hep geometry before? I.e. `m_geoSvc->lcdd()->...`

Reply:
`constantAsString` "hides" the dd4hep access, otherwise we would have to depend explicitly on DD4hep here, wouldn't we?

https://github.com/HEP-FCC/k4SimGeant4/pull/47/files#diff-a7f2b34428c69a67493c4a2c563e87ecf5bdf37f9ff75a2e49fcf2d0c22a2417R107-R109

Reply:
OK. So who is responsible for merging that? Can you assign someone? I am not able

Reply:
@kjvbrt will have a look next week